02-peer-protocol.md

```diff
@@ -390,6 +400,377 @@ would create a Denial of Service risk; therefore, forgetting it is recommended
+for the funding transaction.
+
+
+### The `accept_channel2` Message
+
+This message contains information about a node and indicates its
+acceptance of the new channel.
+
+1. type: 33 (`accept_channel`)
+2. data:
+  * [`32`:`temporary_channel_id`]
+  * [`8`:`dust_limit_satoshis`]
+  * [`8`:`max_htlc_value_in_flight_msat`]
+  * [`8`:`channel_reserve_satoshis`]
+  * [`8`:`htlc_minimum_msat`]
+  * [`4`:`minimum_depth`]
+  * [`2`:`to_self_delay`]
+  * [`2`:`max_accepted_htlcs`]
+  * [`33`:`funding_pubkey`]
+  * [`33`:`revocation_basepoint`]
+  * [`33`:`payment_basepoint`]
+  * [`33`:`delayed_payment_basepoint`]
+  * [`33`:`htlc_basepoint`]
+  * [`33`:`first_per_commitment_point`]
+  * [`2`:`opening_tlv_len`]
+  * [`opening_tlv_len`:`opening_tlv`]
+
+Rationale and Requirements are the same as listed above,
+for [`accept_channel`](#the-accept_channel-message)
+
+### The `funding_compose` Message
+
+This message exchanges the transaction input and output
+information necessary to compose the funding transaction.
+
+1. type: 58 (`funding_compose`)
+2. data:
+  * [`32`:`temporary_channel_id`]
+  * [`2`:`num_inputs`]
+  * [`num_inputs`:`input_info`]
+  * [`2`:`num_outputs`]
+  * [`num_outputs`:`output_info`]
+
+1. subtype `input_info`
+2. data:
+  * [`8`:`satoshis`]
+  * [`32`:`prevtxid`]
+  * [`4`:`prevtxoutnum`]
+  * [`2`:`max_witness_len`]
+  * [`2`:`scriptlen`]
+  * [`scriptlen`:`script`]
+
+1. subtype `output_info`
+2. data:
+  * [`8`:`satoshis`]
+  * [`2`:`scriptlen`]
+  * [`scriptlen`:`script`]
+
+#### Requirements
+
+The sending node:
+  - MUST ensure each `input_info` refers to an existing UTXO
+  - MUST ensure the `output_info`.`script` is a standard script
+  - MUST NOT spend the UTXOs it has specified (until/unless
+    channel establishment fails).
+  - MUST NOT include the channel funding output.
+  - if is the `opener`:
+    - MUST NOT send zero inputs (`num_inputs` cannot be zero).
+    - MUST specify an output with value zero, which will be used
+      as the change address.
+  - if is the `accepter`:
+    - consider the `put_limit` the total of `num_inputs` plus
+      `num_outputs` from `funding_compose`, with minimum 2.
+    - MUST NOT send a number of `input_info` or `output_info` which
+      exceeds the `put_limit`.
+    - MAY send zero inputs and/or outputs.
+
+The receiving node:
+  - if is the `opener`:
+    - MUST fail the channel if:
+      - the sum of `input_info` and `output_info` is greater than
+        the `put_limit`.
+  - otherwise:
+    - MUST fail the channel if:
+      - `change_satoshis` would be negative.
+  - if has not yet sent a `funding_compose`:
+    - MUST send their `funding_compose` message.
+  - otherwise:
+    - MUST use the sent and received `input_info` and `output_info`
+      to create the funding transaction, using `max_extra_witness_len`
+      for each `input_info` and `feerate_per_kw_funding` as specified in
+      `open_channel2`.
+    - MUST send `commitment_signed2`.
```

Comment:
Can we just use `commitment_signed`.  It's identical, and performs the identical function, so I really think it deserves the same name...

Reply:
Right. The rationale was that it might be nice to decouple the two flows as that would allow future modification without needing to worry much about backwards compatibility.